Content.Server/Mobs/Mind.cs

```diff
@@ -272,8 +275,15 @@ public void TransferTo(IEntity? entity)
             OwnedComponent = component;
             OwnedComponent?.InternalAssignMind(this);
 
-            if (VisitingEntity?.HasComponent<GhostComponent>() == false)
+            GhostComponent? ghostComponent = null;
+            VisitingEntity?.TryGetComponent(out ghostComponent);
+
+            if (ghostComponent == null || // visiting entity is not a Ghost
+                (ghostComponent != null && !ghostComponent.CanReturnToBody) || // it is a ghost, but cannot return to body anyway, so it's okay
+                ghostCheckOverride) // to force mind transfer, for example from ControlMobVerb
```

Comment:
my best idea was to add an override for the ghost check + add a check if the ghost can actually return to it's body. With this 2 pieces of information it seems to work. Unless I'm missing something.